test/integration_test/applicationbackup_test.go

```diff
@@ -0,0 +1,472 @@
+	newSecretObj.ResourceVersion = ""
+	newSecret, err := k8s.Instance().CreateSecret(newSecretObj)
+	require.NoError(t, err, "Failed to copy secret %s  to namespace %s", name, namespace)
+
+	backupLocation := &storkv1.BackupLocation{
+		ObjectMeta: meta.ObjectMeta{
+			Name:        name,
+			Namespace:   namespace,
+			Annotations: map[string]string{"stork.libopenstorage.ord/skipresource": "true"},
+		},
+		Location: storkv1.BackupLocationItem{
+			Type:         locationtype,
+			Path:         "test-restore-path",
+			SecretConfig: newSecret.Name,
+		},
+	}
+	_, err = k8s.Instance().CreateBackupLocation(backupLocation)
+	return err
+}
+
+func applicationBackupRestoreTest(t *testing.T) {
+	triggerBackupRestoreTest(
+		t,
+		[]string{testKey + "-backup"},
+		[]string{"mysql-restore"},
+	)
+}
+
+func applicationBackupScheduleTests(t *testing.T) {
+	err := setMockTime(nil)
+	require.NoError(t, err, "Error resetting mock time")
+	t.Run("intervalTest", intervalApplicationBackupScheduleTest)
+	t.Run("dailyTest", dailyApplicationBackupScheduleTest)
+	t.Run("weeklyTest", weeklyApplicationBackupScheduleTest)
+	t.Run("monthlyTest", monthlyApplicationBackupScheduleTest)
+	t.Run("invalidPolicyTest", invalidPolicyApplicationBackupScheduleTest)
+}
+func deletePolicyAndApplicationBackupSchedule(t *testing.T, namespace string, policyName string, applicationBackupScheduleName string) {
+	err := k8s.Instance().DeleteSchedulePolicy(policyName)
+	require.NoError(t, err, fmt.Sprintf("Error deleting schedule policy %v", policyName))
+
+	err = k8s.Instance().DeleteApplicationBackupSchedule(applicationBackupScheduleName, namespace)
+	require.NoError(t, err, fmt.Sprintf("Error deleting applicationBackup schedule %v from namespace %v",
+		applicationBackupScheduleName, namespace))
+
+	time.Sleep(10 * time.Second)
+	applicationBackupList, err := k8s.Instance().ListApplicationBackups(namespace)
+	require.NoError(t, err, fmt.Sprintf("Error getting list of applicationBackups for namespace: %v", namespace))
+	require.Equal(t, 0, len(applicationBackupList.Items), fmt.Sprintf("All applicationBackups should have been deleted in namespace %v", namespace))
+}
+
+func intervalApplicationBackupScheduleTest(t *testing.T) {
+	backupLocation := "backuplocation"
+
+	ctx := createApp(t, "interval-appbackup-sched-test")
+
+	// Create backuplocation here programatically using config-map that contains name of secrets to be used, passed from the CLI
+	err := createBackupLocation(t, backupLocation, ctx.GetID(), storkv1.BackupLocationS3, "secret-config")
+	require.NoError(t, err, "Error creating backuplocation")
+
+	policyName := "intervalpolicy-appbackup"
+	retain := 2
+	interval := 2
+	_, err = k8s.Instance().CreateSchedulePolicy(&storkv1.SchedulePolicy{
+		ObjectMeta: meta.ObjectMeta{
+			Name: policyName,
+		},
+		Policy: storkv1.SchedulePolicyItem{
+			Interval: &storkv1.IntervalPolicy{
+				Retain:          storkv1.Retain(retain),
+				IntervalMinutes: interval,
```

Comment:
isn't allowed minimal interval 15/60 mins ?

Reply:
The minimum is 1 minute actually